src/ConnectedMode.UnitTests/Suppressions/ClientSuppressionSynchronizerTests.cs

```diff
@@ -0,0 +1,132 @@
+        }
+
+        [TestMethod]
+        public void SynchronizeIssues_IssueGetsSuppressedOnServer_LocalIssueGetUpdated()
+        {
+            var issue = CreateIssue(isSuppressedLocally: false);
+
+            var localIssues = new[] { issue };
+            var clientIssueStore = CreateClientIssueStore(localIssues);
+
+            var matches = new[] { issue };
+            var issueFilter = CreateIssuesFilter(localIssues, matches);
+
+            var testSubject = new ClientSuppressionSynchronizer(clientIssueStore, issueFilter);
+
+            issue.IsSuppressed.Should().BeFalse();
+
+            testSubject.SynchronizeSuppressedIssues();
+
+            issue.IsSuppressed.Should().BeTrue();
+        }
+
+        [TestMethod]
+        public void SynchronizeIssues_IssueGetsUnsuppressedOnServer_LocalIssueGetUpdated()
+        {
+            var issue = CreateIssue(isSuppressedLocally: true);
+
+            var localIssues = new[] { issue };
+            var clientIssueStore = CreateClientIssueStore(localIssues);
+            var issueFilter = CreateIssuesFilter(localIssues, new List<IAnalysisIssueVisualization>());
+
+            var testSubject = new ClientSuppressionSynchronizer(clientIssueStore, issueFilter);
+
+            issue.IsSuppressed.Should().BeTrue();
+
+            testSubject.SynchronizeSuppressedIssues();
+
+            issue.IsSuppressed.Should().BeFalse();
+        }
+
+        [TestMethod]
+        public void SynchronizeIssues_MultipleIssuesChanged_LocalIssuesGetUpdated()
+        {
+            var issue1 = CreateIssue(isSuppressedLocally: false);
+            var issue2 = CreateIssue(isSuppressedLocally: true);
+            var issue3 = CreateIssue(isSuppressedLocally: false);
+
+            var localIssues = new[] { issue1, issue2, issue3 };
+            var clientIssueStore = CreateClientIssueStore(localIssues);
+            var matches = new[] { issue1 };
+            var issueFilter = CreateIssuesFilter(localIssues, matches);
+
+            var testSubject = new ClientSuppressionSynchronizer(clientIssueStore, issueFilter);
+
+            issue1.IsSuppressed.Should().BeFalse();
+            issue2.IsSuppressed.Should().BeTrue();
+            issue3.IsSuppressed.Should().BeFalse();
+
+            testSubject.SynchronizeSuppressedIssues();
+
+            issue1.IsSuppressed.Should().BeTrue();
+            issue2.IsSuppressed.Should().BeFalse();
+            issue3.IsSuppressed.Should().BeFalse();
+        }
+
+        private static IClientIssueStore CreateClientIssueStore(params IAnalysisIssueVisualization[] localIssues)
+        {
+            var clientIssueStore = new Mock<IClientIssueStore>();
+
+            clientIssueStore.Setup(x => x.Get()).Returns(localIssues);
+
+            return clientIssueStore.Object;
+        }
+
+        private static IIssuesFilter CreateIssuesFilter(IEnumerable<IAnalysisIssueVisualization> localIssues, IEnumerable<IFilterableIssue> matches = null)
```

Comment:
Did you decide not to use `params` here too?

(there is as slight difference: if you don't pass anything to a params variable it will be set to an empty array rather than null, but that may not matter here).

Reply:
But I can only use params once no? I have 2 lists, that's also why I dont have params for the other method as I have to make the list regardless or maybe I am misunderstanding this

Reply:
Ah, yes; you could make one the arguments `params`, which would make passing one of the list simpler. But not both.